clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerMetricExecutor.java

```diff
@@ -29,16 +30,15 @@
 import org.wildfly.clustering.service.PassiveServiceSupplier;
 
 /**
- * A handler for cache-container metrics.
+ * Executor for cache-container metrics.
  *
  * @author Paul Ferraro
  */
 public class CacheContainerMetricExecutor implements MetricExecutor<EmbeddedCacheManager> {
 
     @Override
     public ModelNode execute(OperationContext context, Metric<EmbeddedCacheManager> metric) throws OperationFailedException {
-        String containerName = context.getCurrentAddressValue();
-        CacheContainer container = new PassiveServiceSupplier<CacheContainer>(context.getServiceRegistry(true), InfinispanRequirement.CONTAINER.getServiceName(context, containerName)).get();
+        CacheContainer container = new PassiveServiceSupplier<CacheContainer>(context.getServiceRegistry(true), InfinispanRequirement.CONTAINER.getServiceName(context, UnaryCapabilityNameResolver.DEFAULT)).get();
```

Comment:
context.getServiceRegistry(true)

That 'true' is nasty because it means what should be a read requires the exclusive lock.

I see that 'true' in a number of places in this PR, and its not a new thing. I'm just kind of randomly choosing this one to comment on.

We might need to make an exception to the rules for this PassiveServiceSupplier thing, although that worries me.

Reply:
We use OperationContext.getServiceRegistry(true) because we are installing (and eventually removing) a temporary service that requires the target service so that we can obtain the requisite Supplier<T>.  If there is a better way to do this, I'm all ears.

Reply:
OK - I've updated this PR to use OperationContext.getServiceRegistry(false) where appropriate.